scripts/main.js

```diff
@@ -78,462 +78,459 @@ let keysObj2 = {
+    let keysContainer1 = document.createElement("div");
+    keysContainer1.classList.add("keysContainer1");
+    keysContainer1.style.display = defDsply1;
+    keysContainer1.style.flexFlow = "row wrap";
+    keysContainer1.style.justifyContent = "center";
+    keysContainer1.style.alignItems = "center";
+    keyboardLayout.appendChild(keysContainer1);
+
+    let keysContainer2 = document.createElement("div");
+    if (keyboardArr2 !== null) {
+      keysContainer2.classList.add("keysContainer2");
+      keysContainer2.style.display = defDsply2;
+      keysContainer2.style.flexFlow = "row wrap";
+      keysContainer2.style.justifyContent = "center";
+      keysContainer2.style.alignItems = "center";
+      keyboardLayout.appendChild(keysContainer2);
+    }
+
+    // SPECIAL BUTTONS INJECTOR
+    let btnInjector = (array) => {
+
+      // button objects with special functions
+      const capsLock = {
+        value: "",
+        style: {
+          width: "13%"
+        },
+        class: {
+          0: "fal",
+          1: "fa-arrow-alt-up",
+        },
+        operator: () => {
+          let btn = document.querySelectorAll(".keysContainer1 button");
+          let indicatorBtn = document.querySelector(".fa-arrow-alt-up div");
+
+          for (let i = 0; i < btn.length; i++) {
+
+            if (/[a-zA-Z]/.test(btn[i].innerText) && btn[i].innerText != switcher2) {
+              let isLowerCase = true;
+
+              if (/[a-z]/.test(btn[i].innerHTML)) {
+                btn[i].innerText = btn[i].innerText.toUpperCase();
+                indicatorBtn.style.background = " rgb(16, 230, 16)";
+                isLowerCase = false;
+              }
+
+              if (isLowerCase) {
+                indicatorBtn.style.background = "rgba(218, 218, 218, 0.6)";
+                btn[i].innerText = btn[i].innerText.toLowerCase();
+              }
+            }
+          }
+        }
+      }
+
+      const backSpace = {
+        value: "",
+        style: {
+          width: "13%"
+        },
+        class: {
+          0: "fal",
+          1: "fa-backspace"
+        },
+        operator: () => {
+          console.clear();
+
+          // if caret already at 0 index, do NOTHING!
+          if (input.selectionStart === 0) return;
+
+          // get access to the input value
+          let txt = input.value;
+
+          // set start index conditionally:
+          //  one position behind the caret if no selection is made, or
+          //  no modification, if a selection range is made
+          let startIdx =
+            input.selectionStart === input.selectionEnd
+              ? input.selectionStart - 1
+              : input.selectionStart;
+
+          let endIdx = input.selectionEnd;
+
+          // declare a "pre-selection" slice of text, a "middle" which will be the deleted portion, and a "post-selection" slice of text
+          let pre = txt.slice(0, startIdx);
+          let mid = txt.slice(startIdx, endIdx);
```

Comment:
I couldn't find the usage of this variable. Could you let me know where it is or will be used?

Reply:
You're right, "mid" is not used....it was a habit to include it.  deleted.